### ISSUE TITLE
Compiled archives for templates with formulas are broken

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/cicero#579]
**Describe the bug**
When using a "compiled" archive for a template with a formula, `cicero` complains that the formula code cannot be compiled.

**To Reproduce**
1. Go to the public template library, to the `fixed-interests` template
2. Compile it using `cicero archive --target cicero`
3. Try and use that template e.g., `cicero parse fixed-interests@0.5.0.cta --sample text/sample.md`
4. See error:
```
bash-3.2$ cicero parse fixed-interests@0.5.0.cicero.cta --sample text/sample.md 
cicero parse fixed-interests@0.5.0.cicero.cta --sample text/sample.md 
12:58:14 PM - error: Compilation error (at file formula_d02c8642fa12d6ed08dea71f0af7a77b0c7893804d0b43b537eb18ea6f666463 line 1 col 1). Cannot find function with name 'monthlyPaymentFormula'
 monthlyPaymentFormula(loanAmount,rate,loanDuration) as "K0,0.00" 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              
CompilerException: Compilation error (at file formula_d02c8642fa12d6ed08dea71f0af7a77b0c7893804d0b43b537eb18ea6f666463 line 1 col 1). Cannot find function with name 'monthlyPaymentFormula'
 monthlyPaymentFormula(loanAmount,rate,loanDuration) as "K0,0.00" 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              
    at Function._throwCompilerException (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/ergo-compiler/lib/scriptmanager.js:292:19)
    at ScriptManager.compileLogic (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/ergo-compiler/lib/scriptmanager.js:324:31)
    at LogicManager.compileLogicSync (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/ergo-compiler/lib/logicmanager.js:266:48)
    at Template.validate (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/cicero-core/lib/template.js:73:28)
    at /Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/cicero-core/lib/templateloader.js:140:16
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/cicero-core/lib/templateloader.js:16:103)
    at _next (/Users/jeromesimeon/.nvm/versions/node/v10.22.0/lib/node_modules/@accordproject/cicero-cli/node_modules/@accordproject/cicero-core/lib/templateloader.js:18:194)
```

**Expected behavior**
It should parse/draft etc as other compiled templates.

**Additional context**

- Unclear if it's a new issue, could well be something already present in `0.20`.
- Problem is that we are calling the Ergo compiler for the formulas even if the template has already been compiled (but for a compiled archive the main code is missing since it has been compiled already)

**Comment by DianaLease** [comment]
@algomaster99 This is what you saw!

**Comment by jeromesimeon** [comment]
> @algomaster99 This is what you saw!

We haven’t used compilednarchives all that much before so I think we are seeing some undetected issues there. Will work on this one for the next fixpack.

**Comment by algomaster99** [comment]
@jeromesimeon cool, makes sense

**Comment by jeromesimeon** [comment]
Fixed in #580
